client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -191,20 +193,21 @@ const designSetup: Step = function DesignSetup( { navigation } ) {
 	}
 
 	function upgradePlan() {
-		// disable for now
-		/*
-		const params = new URLSearchParams( window.location.search );
-		params.append( 'products', PLAN_PREMIUM );
-		history.replace( { search: params.toString() } );
-		*/
+		goToCheckout();
 	}
 
-	function renderCheckoutModal() {
+	function goToCheckout() {
 		if ( ! isEnabled( 'signup/design-picker-premium-themes-checkout' ) ) {
 			return null;
 		}
+		if ( siteSlug ) {
+			const params = new URLSearchParams();
+			params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
 
-		return <AsyncCheckoutModal />;
+			window.location.href = `/checkout/${ encodeURIComponent(
+				siteSlug
+			) }/premium?${ params.toString() }`;
```

Comment:
The "premium" will need to be replaced with the new Pro plan, similar to what was done in importers flow (pdgrnI-Be-p2). Context pdgrnI-xR-p2